desc/objectives/objective_funs.py

```diff
@@ -474,8 +553,6 @@ def jac_scaled_error(self, x, constants=None):
 
         if self._deriv_mode == "batched":
             J = Derivative(self.compute_scaled_error, mode="fwd")(x, constants)
-        if self._deriv_mode == "looped":
```

Comment:
This is a completely unrelated comment but I think we should explain this jacobian, jvp, etc better in some dev guide. For example, an individual objectives `jac_scaled_error` doesn't use `batched_vmap` it is usually `jax.jacfwd`. On the other hand, when we wrap the objective and constraints, the jacobian is calculated by corresponding `jvp_` method, not `jac_` method. Even now it confuses me. What is the case we use `jac_` method instead of `jvp_`? Good thing to clarify in long-waiting dev-guide